pkg/operator/controller/ingress-routes/controller.go

```diff
@@ -0,0 +1,250 @@
+				return reconcile.Result{Requeue: true}, err
+			}
+
+			continue
+		}
+
+		if err := r.ensureServiceCertKeyPairSecretRole(ingress, componentRoute.Name, secretName); err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+
+		if err := r.ensureServiceCertKeyPairSecretRoleBinding(ingress, componentRoute.Name, componentRoute.ConsumingUsers); err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) ensureServiceCertKeyPairSecretRole(owner metav1.Object, crName, secretName string) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			// TODO: Generate Name and add labels to avoid componentRoute.Name collisions.
+			Name:      crName,
+			Namespace: r.config.SecretNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "config.openshift.io/v1",
+					Kind:       "Ingress",
+					Name:       owner.GetName(),
+					UID:        owner.GetUID(),
+				},
+			},
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"get", "list"},
+				APIGroups:     []string{""},
+				Resources:     []string{"secrets"},
+				ResourceNames: []string{secretName},
+			},
+		},
+	}
+
+	// Attempt to create or update the roleBinding
+	if err := r.client.Create(context.TODO(), role); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return err
+		}
+
+		existingRole := &rbacv1.Role{}
+		if err := r.client.Get(context.TODO(), client.ObjectKey{Name: role.GetName(), Namespace: role.GetNamespace()}, existingRole); err != nil {
+			return err
+		}
+
+		// TODO: Remove comment below once decision is made.
+		// Hard update of spec - could apply hash annotation of desired spec
+		// to role to avoid race conditions with other controllers.
+		existingRole.OwnerReferences = role.OwnerReferences
+		existingRole.Rules = role.Rules
+		return r.client.Update(context.TODO(), existingRole)
+	}
+
+	return nil
+}
+
+func (r *reconciler) ensureServiceCertKeyPairSecretRoleBinding(owner metav1.Object, roleName string, users []string) error {
+	subjects := []rbacv1.Subject{}
+	for _, user := range users {
+		subjects = append(subjects, rbacv1.Subject{
+			Kind:     rbacv1.ServiceAccountKind, // TODO: Should we support non service accounts?
+			Name:     user,
+			APIGroup: "",
+		})
+	}
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleName,
+			Namespace: r.config.SecretNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "config.openshift.io/v1",
+					Kind:       "Ingress",
+					Name:       owner.GetName(),
+					UID:        owner.GetUID(),
+				},
+			},
+		},
+		Subjects: subjects,
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     roleName,
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	// Attempt to create or update the roleBinding
+	if err := r.client.Create(context.TODO(), roleBinding); err != nil {
```

Comment:
same library-go comment

Reply:
Addressed in the latest version of the PR.